api/helpers/gamestate/moves/resolve/validate.js

```diff
@@ -0,0 +1,58 @@
+const GamePhase = require('../../../../../utils/GamePhase.json');
+
+module.exports = {
+  friendlyName: 'Validate request to resolve pending one-off and counters',
+
+  description: 'Verifies whether a request to resolve one-offs is legal, throwing explanatory error if not.',
+
+  inputs: {
+    currentState: {
+      type: 'ref',
+      descriptions: 'Object containing the current game state',
+      required: true,
+    },
+    /**
+     * @param { MoveType.RESOLVE } requestedMove.moveType - Specifies that this a resolve move
+     * @param { Object } requestedMove - Object describing the request to resolve
+     * @param { 1 | 0 } requestedMove.playedBy - Which player is resolving
+     */
+    requestedMove: {
+      type: 'ref',
+      description: 'Object containing data needed for current move',
+      required: true,
+    },
+    playedBy: {
+      type: 'number',
+      description: 'Player number of player requesting move',
+      required: true,
+    },
+  },
+  sync: true,
+  fn: ({ currentState, playedBy }, exits) => {
+    try {
+
+      // Must be COUNTERING phase
+      if (currentState.phase !== GamePhase.COUNTERING) {
+        throw new Error(`Can only resolve during the countering phase`);
+      }
+
+      if (!currentState.oneOff) {
+        throw new Error('You cannot resolve unless there is a one-off pending');
+      }
+
+      // Must be your chance to counter
+      // It's your chance if it is your turn and there are an odd number of twos
+      // or if it's your opponent's turn and there are an even number of twos
+      // (since the opponent of the person who played the one-off counters first)
+      const isYourTurn = currentState.turn % 2 === playedBy;
+      const numTwosIsEven = currentState.twos.length % 2 === 0;
+      if (isYourTurn === numTwosIsEven) {
+        throw new Error('Waiting for opponent to counter');
+      }
```

Comment:
This took me a while to suss out, so feedback is welcome on the clarity. Whether you are the one who can counter or not right now comes down to whether it's your turn and how many twos there are.

If it is your turn, then you played the one-off and then your opponent should counter first, which means you can only play if there are an odd number of twos. So it's an error if it's your turn and there are an even number of twos i.e. `isYourTurn === numTwosIsEven === true`

If it is **not** your turn, then your opponent played the one-off and _you_ counter first, so it is an error if it is not your turn and there are an odd number of twos i.e. `isYourTurn === numTwosIsEven === false`.

So ultimately that means we should error if both values are true or if both values are false, which we can represent succinctly as error if both values equal each other. It took me a while to work that out, so again, @seriouslysean and @Haviles04 I'd love to hear your thoughts on how it reads.

@Haviles04 this exact logic is also needed to verify if you can play a counter for #952 so we can discuss whether it makes sense to abstract it into a helper